src/main/java/org/prebid/server/auction/privacycontextfactory/AmpPrivacyContextFactory.java

```diff
@@ -0,0 +1,135 @@
+
+        return tcfDefinerService.resolveTcfContext(
+                strippedPrivacy,
+                resolveAlpha2CountryCode(device),
+                resolveIpAddress(device, strippedPrivacy),
+                extractGdprConfig(account),
+                requestType,
+                requestLogInfo(requestType, bidRequest, account.getId()),
+                timeout)
+                .map(tcfContext -> logWarnings(auctionContext, tcfContext))
+                .map(tcfContext -> PrivacyContext.of(strippedPrivacy, tcfContext, tcfContext.getIpAddress()));
+    }
+
+    private Privacy stripPrivacy(Privacy privacy, AuctionContext auctionContext) {
+        final List<String> errors = auctionContext.getPrebidErrors();
+
+        final String consentTypeParam = auctionContext.getHttpRequest().getQueryParams().get(CONSENT_TYPE_PARAM);
+        final ConsentType consentType = ConsentType.from(consentTypeParam);
+
+        if (consentType == ConsentType.unknown) {
+            errors.add("Invalid consent_type param passed");
+            return privacy.withoutConsent();
+        } else if (consentType == ConsentType.tcfV1) {
+            errors.add("Consent type tcfV1 is no longer supported");
+            return privacy.withoutConsent();
+        }
+
+        return privacy;
+    }
+
+    private AccountGdprConfig extractGdprConfig(Account account) {
+        final AccountPrivacyConfig accountPrivacyConfig = account != null ? account.getPrivacy() : null;
+
+        return accountPrivacyConfig != null ? accountPrivacyConfig.getGdpr() : null;
+    }
+
+    private static TcfContext logWarnings(AuctionContext auctionContext, TcfContext tcfContext) {
+        auctionContext.getDebugWarnings().addAll(tcfContext.getWarnings());
+
+        return tcfContext;
+    }
+
+    private String resolveAlpha2CountryCode(Device device) {
+        final Geo geo = device != null ? device.getGeo() : null;
+        final String alpha3CountryCode = geo != null ? geo.getCountry() : null;
+
+        return countryCodeMapper.mapToAlpha2(alpha3CountryCode);
+    }
+
+    private String resolveIpAddress(Device device, Privacy privacy) {
+        final boolean shouldBeMasked = Objects.equals(privacy.getCoppa(), 1)
+                || (device != null && Objects.equals(device.getLmt(), 1));
+
+        final String ipV4Address = device != null ? device.getIp() : null;
+        if (StringUtils.isNotBlank(ipV4Address)) {
+            return shouldBeMasked ? ipAddressHelper.maskIpv4(ipV4Address) : ipV4Address;
+        }
+
+        final String ipV6Address = device != null ? device.getIpv6() : null;
+        if (StringUtils.isNotBlank(ipV6Address)) {
+            return shouldBeMasked ? ipAddressHelper.anonymizeIpv6(ipV6Address) : ipV6Address;
+        }
+
+        return null;
+    }
+
+    private static RequestLogInfo requestLogInfo(MetricName requestType, BidRequest bidRequest, String accountId) {
+        if (Objects.equals(requestType, MetricName.openrtb2web)) {
```

Comment:
As we work with enum value`MetricName.openrtb2web == requestType. Pls make same change for enforcement service